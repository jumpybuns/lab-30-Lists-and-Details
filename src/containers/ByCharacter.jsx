import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import Character from '../components/list/Character';
import { fetchCharacter } from '../services/heyArnoldAPI';
import PropTypes from 'prop-types';

function ByCharacter() {
  const [character, setCharacter] = useState([]);
  const [loading, setLoading] = useState(true);
  const { name } = useParams();

  useEffect(() => {
    fetchCharacter(name).then((res) => {
      setCharacter(res), setLoading(false);
    });
  }, []);

  if (loading) return <h1>Loading...</h1>;
  return (
    <div>
      <Character character={character} />
    </div>
  );
}

ByCharacter.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      name: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default ByCharacter;
