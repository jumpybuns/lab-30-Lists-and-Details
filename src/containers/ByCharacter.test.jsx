import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import ByCharacter from './ByCharacter';

describe('ByCharacter container', () => {
  it('displays quotes by a character', async () => {
    render(<ByCharacter match={{ params: { character: 'Arnold' } }} />);

    screen.getByText('Loading');

    const ul = await screen.findByTestId('quotes');

    return waitFor(() => {
      expect(ul).not.toBeEmptyDOMElement();
    });
  });
});
