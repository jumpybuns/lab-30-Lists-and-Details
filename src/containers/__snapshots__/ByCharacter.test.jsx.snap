// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ByCharacter component renders ByCharacter 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
